src/lib/comms/sol-network-impl-linux.c

```diff
@@ -464,24 +464,24 @@ sol_network_subscribe_events(void (*cb)(void *data, const struct sol_network_lin
 
     SOL_NULL_CHECK(cb, false);
 
-    if (sol_network_start_netlink() < 0) return false;
+    if (sol_network_start_netlink() < 0) return -errno;
 
     callback = sol_vector_append(&network->callbacks);
     SOL_NULL_CHECK(callback, false);
```

Comment:
You must also change this return value. Replace `false` with `-ENOMEM`
Remenber to check all return values and replace `true` and `false` with the proper values